packages/build/tests/telemetry/tests.js

```diff
@@ -127,38 +135,50 @@ test('Telemetry is disabled by default', async (t) => {
 })
 
 test('Telemetry can be enabled via flag', async (t) => {
-  const { telemetryRequests } = await runWithApiMock(t, 'success', { telemetry: true })
+  const { telemetryRequests } = await runWithApiMock(t, 'success', {
+    telemetry: true,
+    featureFlags: 'buildbot_build_telemetry',
+  })
   t.is(telemetryRequests.length, 1)
 })
 
-test('Telemetry BUILD_TELEMETRY_DISABLED env var overrides flag usage', async (t) => {
+test('Telemetry cannnot be enabled via flag if the feature flag is disabled', async (t) => {
   const { telemetryRequests } = await runWithApiMock(t, 'success', {
     telemetry: true,
-    env: { BUILD_TELEMETRY_DISABLED: 'true' },
+    flags: {},
   })
   t.is(telemetryRequests.length, 0)
 })
 
-test('Telemetry error generates no logs', async (t) => {
-  await runWithApiMock(t, 'success', { origin: 'https://...', telemetry: true, snapshot: true })
+test('Telemetry BUILD_TELEMETRY_DISABLED env var overrides flag and feature flag usage', async (t) => {
+  const { telemetryRequests } = await runWithApiMock(t, 'success', {
+    telemetry: true,
+    featureFlags: 'buildbot_build_telemetry',
+    env: { BUILD_TELEMETRY_DISABLED: 'true' },
+  })
+  t.is(telemetryRequests.length, 0)
 })
 
-test("Telemetry's unref'd child process does not disturb regular build process execution on success", async (t) => {
-  const { exitCode } = await runWithApiMock(t, 'success', {
+test('Telemetry error generates no logs', async (t) => {
+  await runWithApiMock(t, 'success', {
+    origin: 'https://...',
     telemetry: true,
-    telemetryUnrefChildProcess: true,
-    useBinary: true,
+    featureFlags: 'buildbot_build_telemetry',
     snapshot: true,
   })
-  t.is(exitCode, 0)
 })
 
-test("Telemetry's unref'd child process does not disturb regular build process execution on error", async (t) => {
-  const { exitCode } = await runWithApiMock(t, 'invalid', {
-    telemetry: true,
-    telemetryUnrefChildProcess: true,
-    useBinary: true,
-    snapshot: true,
+test('Telemetry calls timeout in less than 3 seconds by default', async (t) => {
+  // Start the mock telemetry server
+  const { scheme: schemeTelemetry, host: hostTelemetry, requests, stopServer } = await startServer({
+    path: TELEMETRY_PATH,
+    // eslint-disable-next-line no-magic-numbers
+    wait: 3000,
```

Comment:
Instead of making the test wait using a hardcoded duration, would it possible to wait for the telemetry failure?

Using timeouts in tests can sometimes randomly fail when running tests in parallel, due to the additional load induced by the other tests running at the same time.

Reply:
See - https://github.com/netlify/build/pull/2438#discussion_r597684106

That wait period right now is kind of tied to the test, but once we have a way to validate the error type we can make the waiting period arbitrarily high, because what we want to test is that the client execution finishes before the server responds. The alternative to passing in this setting would be to actually block the server indefinitely... which sounds a bit weird to me 🤔

Reply:
Yes, that makes sense. :+1: 

Could we do so while still calling `@netlify/build` entry point (instead of importing `trackBuildComplete()` directly)? This would allow re-using the `runWithApiMock()` shared code. Also, this would allow not depending on `trackBuildComplete()` internal implementation, and only test this from a user standpoint.

Reply:
Yep, makes sense, later on we can use the `errorMonitor` mock to validate the `bugsnug` output and see that we're dealing with a timeout, making this test more accurate 👍 

Addressed via https://github.com/netlify/build/pull/2438/commits/88419ae366d83d272229226fa481ce119eb444bf